ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/ProjectAssignment/Index.cshtml.cs

```diff
@@ -0,0 +1,70 @@
+using ApplyToBecome.Data;
+using ApplyToBecome.Data.Models;
+using ApplyToBecome.Data.Services;
+using ApplyToBecome.Data.Services.Interfaces;
+using ApplyToBecomeInternal.Extensions;
+using ApplyToBecomeInternal.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ApplyToBecomeInternal.Pages.ProjectAssignment
+{
+	public class IndexModel : PageModel
+	{
+		private readonly IUserRepository _userRepository;
+		private readonly IAcademyConversionProjectRepository _academyConversionProjectRepository;
+		public IndexModel(IUserRepository userRepository, IAcademyConversionProjectRepository academyConversionProjectRepository)
+		{
+			_academyConversionProjectRepository = academyConversionProjectRepository;
+			_userRepository = userRepository;
+		}
+
+		public string SchoolName { get; private set; }
+		public int Id { get; set; }
+		public IEnumerable<User> DeliveryOfficers { get; set; }
+		public string SelectedDeliveryOfficer { get; set; }
+
+		public async Task<IActionResult> OnGet(int id)
+		{
+			var projectResponse = await _academyConversionProjectRepository.GetProjectById(id);
+			Id = id;			
+			SchoolName = projectResponse.Body.SchoolName;
+			SelectedDeliveryOfficer = projectResponse.Body?.AssignedUser?.FullName;
+
+			DeliveryOfficers = await _userRepository.GetAllUsers();
+
+			return Page();
+		}
+
+		public async Task<IActionResult> OnPost(int id, string selectedName, bool unassignDeliveryOfficer)
+		{
+			if (unassignDeliveryOfficer)
+			{
+				var updatedProject = new UpdateAcademyConversionProject
+				{
+					AssignedUser = new User(Guid.Empty.ToString(), " ", " ")
+				};
+
+				await _academyConversionProjectRepository.UpdateProject(id, updatedProject);
+				TempData.SetNotification(NotificationType.Success, "Done", "Project is unassigned");
+			}
+			else if (!string.IsNullOrEmpty(selectedName))
+			{
+				var deliveryOfficers = await _userRepository.GetAllUsers();
```

Comment:
We already have all the Users in this viewmodel, shouldn't need to call GetAllUsers again.

Reply:
Are you sure about that? My understanding is for a new request we would need to lookup that information again.
I'll test it.

Reply:
I see no reason why the ViewModels DeliveryOfficers list should be any different, but maybe I'm missing something referentially(?).

Reply:
Yea as I thought its null.

![image](https://user-images.githubusercontent.com/109524605/194348931-45ca63ce-7e66-48d0-89a9-cc2789574896.png)

Any information obtained in the get would need to be resubmitted in the post in order to have access to it.

Reply:
Ah good to know, cheers for testing.